bench/micro/Bench/Database/LSMTree/Internal/Lookup.hs

```diff
@@ -10,93 +11,113 @@
 import           System.Random as R
 import           Test.QuickCheck (generate, shuffle)
 
 -- | TODO: add a separate micro-benchmark that includes multi-pages.
 benchmarks :: Benchmark
 benchmarks = bgroup "Bench.Database.LSMTree.Internal.Lookup" [
-      bgroup "prepLookups for a single run" [
-          benchPrepLookups defaultConfig
-        , benchPrepLookups (defaultConfig {
-              name = "default onlyPos"
-            , nneg = 0
-            })
-        , benchPrepLookups (defaultConfig {
-              name = "default onlyNeg"
-            , npos = 0
-            })
-        , benchPrepLookups (defaultConfig {
-              name = "default high fpr"
-            , fpr  = 0.9
-            })
-        ]
+      benchLookups defaultConfig
     ]
-  where
-    benchPrepLookups :: Config -> Benchmark
-    benchPrepLookups conf@Config{name} =
-      env (prepLookupsEnv (Proxy @UTxOKey) conf) $ \ ~(rs, ks) ->
+
+benchLookups :: Config -> Benchmark
+benchLookups conf@Config{name} =
+    withEnv $ \ ~(_dir, _hasFS, hasBlockIO, bsize, rs, ks) ->
+      env ( pure ( V.map Run.runFilter rs
+                 , V.map Run.runIndex rs
+                 , V.map Run.runKOpsFile rs
+                 )
+          ) $ \ ~(blooms, indexes, kopsFiles) ->
         bgroup name [
             -- The bloomfilter is queried for all lookup keys. The result is an
             -- unboxed vector, so only use whnf.
             bench "Bloomfilter query" $
-              whnf (\ks' -> bloomQueriesDefault rs ks') ks
+              whnf (\ks' -> bloomQueriesDefault blooms ks') ks
             -- The compact index is only searched for (true and false) positive
             -- lookup keys. We use whnf here because the result is an unboxed
             -- vector.
-          , env (pure $ bloomQueriesDefault rs ks) $ \rkixs ->
+          , env (pure $ bloomQueriesDefault blooms ks) $ \rkixs ->
               bench "Compact index search" $
-                whnfAppIO (\ks' -> indexSearches rs ks' rkixs) ks
+                whnfAppIO (\ks' -> indexSearches indexes kopsFiles ks' rkixs) ks
             -- All prepped lookups are going to be used eventually so we use
             -- @nf@ on the vector of 'IOOp's. We only evaluate the vector of
             -- indexes to WHNF, because it is an unboxed vector.
-          , bench "In-memory lookup" $
-              nfAppIO (\ks' -> prepLookups rs ks' >>= \(rkixs, ioops) -> pure (rkixs `seq` ioops)) ks
+          , bench "Lookup preparation in memory" $
+              whnfAppIO (\ks' -> prepLookups blooms indexes kopsFiles ks') ks
+          , bench "Batched lookups in IO" $
+              nfAppIO (\ks' -> lookupsInBatches hasBlockIO bsize resolveV rs blooms indexes kopsFiles ks') ks
           ]
+  where
+    withEnv = envWithCleanup
+                (lookupsInBatchesEnv conf)
+                lookupsInBatchesCleanup
+    -- TODO: pick a better value resolve function
+    resolveV = \v1 _v2 -> v1
```

Comment:
I think it's fine to just use `const`, we're not too focused on `Mupdate` performance for now.